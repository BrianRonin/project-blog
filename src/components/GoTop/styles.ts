import styled, { css, DefaultTheme } from 'styled-components'

const Light = (theme: DefaultTheme) => css`
  color: ${theme.colors.primary};
  background: ${theme.colors.white};
`
type background = {
  background: boolean
}

export const Main = styled.a<background>`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
