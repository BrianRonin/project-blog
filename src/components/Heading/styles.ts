import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xlarge};
    @media ${theme.media.medium} {
      font-size: ${theme.sizes.large};
    }
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1<
  Pick<HeadingProps, 'light' | 'size' | 'uppercase'>
>`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primary};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${!!size && titleSize[size](theme)};
  `}
`
