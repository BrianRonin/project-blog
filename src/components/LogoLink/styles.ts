import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    width: 7rem;

    border-radius: 50%;
  }
`

export const Logo = styled.img`
  max-height: 6rem;
`
