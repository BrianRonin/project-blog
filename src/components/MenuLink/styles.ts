import styled, { css } from 'styled-components'
export const Main = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    display: block;
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    font-size: 1.8rem;
    border-right: 0.5rem solid ${theme.colors.primary};

    &:hover {
      color: ${theme.colors.secondary};
      border-right-color: ${theme.colors.secondary};
    }
  `}
`
