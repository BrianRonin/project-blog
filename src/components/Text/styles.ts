import styled, { css } from 'styled-components'
export const Main = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.sizes.medium};
  `}
`
