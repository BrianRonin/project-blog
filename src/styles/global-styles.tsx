import {
  createGlobalStyle,
  css,
} from 'styled-components'
/*
font-family: 'Montserrat', sans-serif;
font-family: 'Open Sans', sans-serif;
font-family: 'Silkscreen', cursive;
*/

export const GlobalStyles = createGlobalStyle`
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
  scroll-behavior: smooth;
}

body {
  ${({ theme }) => css`
    background: ${theme.colors.mainBg};
  `}
  font-size: 16px;
  font-family: ${({ theme }) =>
    theme.fonts.family.default};
  color: ${({ theme }) => theme.colors.darkText}
}

h1, h2, h3, h4, h5, h6 {
  font-family: ${({ theme }) =>
    theme.fonts.family.default};
  margin: ${({ theme }) =>
    theme.spacings.large} 0;
  font-weight: 800;
  color: ${({ theme }) => theme.colors.primary}
}

/* p {
  margin: ${({ theme }) =>
    theme.spacings.large} 0;
}

ul, ol {
  margin: ${({ theme }) => theme.spacings.medium};
  padding: ${({ theme }) =>
    theme.spacings.medium};
}

a {
  color: ${({ theme }) => theme.colors.secondary}
}

table {
  width: 100%;
  overflow-y: auto;
}

code, pre {
  width: 100vw;
} */
`
