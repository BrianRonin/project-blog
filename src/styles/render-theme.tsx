import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from './themes/default'

export const renderTheme = (children: any) => {
  return render(
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>,
  )
}
