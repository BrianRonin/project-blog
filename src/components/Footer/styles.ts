import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    background-color: ${theme.colors.white};
    a {
      color: ${theme.colors.darkText};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`
