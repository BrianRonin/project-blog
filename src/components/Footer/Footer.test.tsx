import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/render-theme'
import { Footer } from './index'

describe('<Footer />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Footer>teste</Footer>)
    expect(screen.getByText('teste')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
