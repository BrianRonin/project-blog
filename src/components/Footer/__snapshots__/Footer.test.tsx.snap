// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c2 {
  font-size: 2.4rem;
}

.c0 {
  text-align: center;
  font-size: 1.6rem;
  border-top: 0.1rem solid #DDDDDD;
  background-color: white;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c0 .c1 {
  font-size: 1.6rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      teste
    </div>
  </div>
</div>
`;
